middleware_auth.go

```diff
@@ -265,26 +265,38 @@ func RequiresPermission(opts RequiresPermissionMiddlewareOpts) Middleware {
 }
 
 func (rp *requiresPermissionHandler) ServeHTTP(rw http.ResponseWriter, r *http.Request, next http.HandlerFunc) {
-	ctx := r.Context()
 	vars := GetVars(r)
-	var resource string
+	var resources []string
 	var status int
 	var err error
 	if rp.opts.ResourceFunc != nil {
-		resource, status, err = rp.opts.ResourceFunc(r)
+		resources, status, err = rp.opts.ResourceFunc(r)
 		if err != nil {
 			http.Error(rw, err.Error(), status)
 			return
 		}
 	} else {
 		for _, level := range rp.opts.ResourceLevels {
 			if resourceVal, exists := vars[level]; exists {
-				resource = resourceVal
+				resources = []string{resourceVal}
 				break
 			}
 		}
 	}
 
+	if len(resources) == 0 {
+		http.Error(rw, "no resources found", http.StatusUnauthorized)
+		return
+	}
+	for _, item := range resources {
+		if ok := rp.handleResource(rw, r.Context(), item); !ok {
+			return
+		}
+	}
+	next(rw, r)
+}
+
+func (rp *requiresPermissionHandler) handleResource(rw http.ResponseWriter, ctx context.Context, resource string) bool {
```

Comment:
Sorry for not spotting this sooner but I think we want to just have this check permissions, and do the user stuff outside the loop. We don't want to check if they're authenticated more than once since it can potentially be expensive

Reply:
good call